CMakeLists.txt

```diff
@@ -386,6 +386,16 @@ include_directories( ${KRATOS_SOURCE_DIR}/external_libraries )
 # defines needed
 add_definitions( -DKRATOS_PYTHON )
 
+# Additional considerations depending of the apps compiled
+if(DEFINED ENV{KRATOS_APPLICATIONS})
+    foreach(APPLICATION_PATH $ENV{KRATOS_APPLICATIONS})
+        get_filename_component(APPLICATION_NAME ${APPLICATION_PATH} NAME)
+        if("FreeSurfaceApplication" MATCHES ${APPLICATION_NAME})
+            add_definitions( -DKRATOS_FREE_SURFACE_APPLICATION_INCLUDED )
+        endif("FreeSurfaceApplication" MATCHES ${APPLICATION_NAME})
+    endforeach(APPLICATION_PATH $ENV{KRATOS_APPLICATIONS})
+endif(DEFINED ENV{KRATOS_APPLICATIONS})
```

Comment:
:flushed: :flushed: :flushed: 


![image](https://user-images.githubusercontent.com/25484702/78922708-4b49ce80-7a97-11ea-9185-31bb30acadba.png)

Reply:
Any other suggestion? (since we change the compilation mechanism I am little bit lost)

Reply:
@roigcarlo

Reply:
No sorry we can't do that. If this is the only solution its best to have it in the .h

Reply:
> Any other suggestion? (since we change the compilation mechanism I am little bit lost)

There is none, thats how templates work...
Thats why there is no (std::)vector.cpp

=> has to remain header-only

Reply:
OK, moving back...

Reply:
I am reverting the source file, but IMO we can get rid of the templates in a future (these classes could be considered in the constructor directly)